MetaMask/metamask-mobile
# Use converted input value if internal currency is FIAT

**Description**

This PR adds the following changes to Amount screen:

- Inserted amount is now validated correctly when Fiat is selected as primary currency in Settings;
- It should now provide converted Fiat values to all tokens that have conversion rate available, not only for mainnet;
- Improve safety while converting `number` to `BN`;
- Fix inconsistent decimal cases between native tokens and ERC-20 tokens when converted to Fiat;

**What to test**

- ***Case 1 - Insufficient funds fix***
   - Switch to "Binance Smart Chain Mainet"
   - Switch to "Fiat" as primary currency under "Settings > General"
   - Go to Send view
   - Send a value under your balance, it should proceed.

- ***Case 2 - Insufficient funds fix***
   - Switch to "Binance Smart Chain Mainet"
   - Switch to "Fiat" as primary currency under "Settings > General"
   - Go to Send view
   - Send a value above your balance, it should error "Insufficient funds".

- ***Case 3 - Convert FIAT value for tokens (non-ETH)***
   - Switch to "Binance Smart Chain Mainet" (other networks can also be tested)
   - Switch to "Fiat" as primary currency under "Settings > General"
   - Go to send view
   - Insert FIAT value
   - It should show converted BNB value

Note: To validate correct conversion values please compare it with [cryptocompare](https://www.cryptocompare.com/)

***Screenshot***
<img width="394" alt="Screen Shot 2022-12-28 at 16 10 09" src="https://user-images.githubusercontent.com/575479/209840562-14cddd78-3318-4cfc-b2c3-99c7357ecbc0.png">

**Checklist**


**Issue**

Resolves #3983
